utilities/src/serialize/impls.rs

```diff
@@ -437,6 +437,47 @@ impl<T: CanonicalDeserialize> CanonicalDeserialize for Vec<T> {
     }
 }
 
+impl<T: CanonicalDeserialize + std::fmt::Debug> CanonicalDeserialize for [T; 32] {
+    #[inline]
+    fn deserialize_with_mode<R: Read>(
+        mut reader: R,
+        compress: Compress,
+        validate: Validate,
+    ) -> Result<Self, SerializationError> {
+        let mut values = [(); 32].map(|_| T::deserialize_with_mode(&mut reader, compress, Validate::No));
```

Comment:
```suggestion
        let mut values = (0..32).map(|_| T::deserialize_with_mode(&mut reader, compress, Validate::No));
```

Reply:
also, I see no need for mutability here

Reply:
scratch that out; I see that this is a trick to have an array instead of just an iterator, and that it's reused